sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -1108,35 +1110,31 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
   }
 
   test("dynamic partition value test") {
-    try {
-      sql("set hive.exec.dynamic.partition.mode=nonstrict")
-      // date
-      sql("drop table if exists dynparttest1")
-      sql("create table dynparttest1 (value int) partitioned by (pdate date)")
-      sql(
-        """
-          |insert into table dynparttest1 partition(pdate)
-          | select count(*), cast('2015-05-21' as date) as pdate from src
-        """.stripMargin)
-      checkAnswer(
-        sql("select * from dynparttest1"),
-        Seq(Row(500, java.sql.Date.valueOf("2015-05-21"))))
+    withTable("dynparttest1", "dynparttest2") {
+      withSQLConf("hive.exec.dynamic.partition.mode" -> "strict") {
+        sql("set hive.exec.dynamic.partition.mode=nonstrict")
```

Comment:
This behave different now？

Reply:
nice catch. updated

Reply:
It still seems wrong to me. You should test with an inner wrapper using `withSQLConf + nonstrict` and restore the `strict` mode with an outer wrapper `try-finally`.

Reply:
That is the purpose of using `withSQLConfig`. So each test case should set the configs it needs and restores only to default values.

A test case should not pollute other test cases nor care about starting values of confs of other test cases.

Reply:
So replacing `nonstrict` with `strict` is an intentional change?

Reply:
So previous code does something like
1. set the mode
2. the recover it by the finally so other test cases are not impacted.

The new code also set the mode, and recover it by `withSQLConf`

Reply:
https://github.com/apache/spark/pull/46778#discussion_r1618216246 move the following questions there